parser-typechecker/unison/Main.hs

```diff
@@ -114,56 +121,61 @@ main = do
            _                                 -> (Nothing, args)
   currentDir <- getCurrentDirectory
   configFilePath <- getConfigFilePath mcodepath
+  config@(config_, _cancelConfig) <-
+    catchIOError (watchConfig configFilePath) $ \_ ->
+      Exit.die "Your .unisonConfig could not be loaded. Check that it's correct!"
+  branchCacheSize :: Word <- Config.lookupDefault 4096 config_ "NamespaceCacheSize"
+  branchCache <- Cache.semispaceCache branchCacheSize
```

Comment:
We moved config initialization here, since the branchCache needs to be created here, and we wanted a config setting for the branchCache (the `NamespaceCacheSize` setting).

Reply:
Maybe we should thread a UnisonConfig monad class throughout the app so we can access it in `FileCodebase.hs`, etc.

Reply:
Sounds intriguing... for later.